realm/realm-library/src/objectServer/java/io/realm/ClientResetObjectServerError.java

```diff
@@ -21,16 +21,16 @@
 /**
  * Class encapsulating information needed for handling a Client Reset event.
  *
- * @see io.realm.SyncSession.ErrorHandler#onClientResetRequired(SyncSession, ClientResetHandler) for more information
+ * @see io.realm.SyncSession.ErrorHandler#onError(SyncSession, ObjectServerError) for more information
  *      about when and why Client Reset occurs and how to deal with it.
  */
-public class ClientResetHandler extends ObjectServerError {
+public class ClientResetObjectServerError extends ObjectServerError {
```

Comment:
I'm not sure how I feel about this name. It seems overly long. Also consider the class has a method named `executeClientReset()`.

Some other suggestions:
- `ClientResetRequiredError`
- `ClientResetError`

Thoughts?